sql/sql_repl.cc

```diff
@@ -411,7 +411,7 @@ inline void fix_checksum(String *packet, ulong ev_offset)
               BINLOG_CHECKSUM_ALG_DESC_LEN + BINLOG_CHECKSUM_LEN);
   crc= my_checksum(0, (uchar *)packet->ptr() + ev_offset, data_len -
                    BINLOG_CHECKSUM_LEN);
-  int4store(packet->ptr() + ev_offset + data_len - BINLOG_CHECKSUM_LEN, crc);
+  int4store(&(*packet)[0] + ev_offset + data_len - BINLOG_CHECKSUM_LEN, crc);
```

Comment:
Why?

Reply:
Same as above, `ptr()` is `const` here.

Reply:
May be `const_cast` instead? What you do looks quite fragile.

Reply:
Giving the code for both method I do not understand what fragility do you mean here. I don't think it's most fragile here. `const_cast` lets us avoid type system requirements and this probably not good.

Reply:
Sorry, I somehow missed that `String` has overloaded `[]`. Still looks rather scary.

Reply:
Will adding non-const `ptr()` be better than that?